### PR TITLE
MODUSERS-114: Log exception including stacktrace for streaming GET Users

### PR DESCRIPTION
**Pull request by julianladisch**
Pass JsonProcessingException to reply handler to return a failure.

Do not swallow the exception stacktrace, log it.

Use English or the tenant language or the language requested by the client, but
not the language of the server.

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/UsersAPI.java [hunk 1]
```diff
@@ -183,13 +184,14 @@ public void getUsers(String query, String orderBy,
               try {
                 response.write(JSON_MAPPER.writeValueAsString(user), "UTF-8");
               } catch (JsonProcessingException e) {
-                logger.error("This error should never happen: " + e);
+                throw new UncheckedIOException(e);
```
Comment (hjiebsco) [inline src/main/java/org/folio/rest/impl/UsersAPI.java]: User json -> User in RMB, and then -> User json here again. How likely is the exception? But I am fine with re-throw.

File: src/main/java/org/folio/rest/impl/UsersAPI.java [hunk 2]
```diff
@@ -183,13 +184,14 @@ public void getUsers(String query, String orderBy,
...
               }
             }, reply -> {
               if (reply.succeeded()) {
                 response.write(String.format(JSON_USERS_FOOTER, cnt[0], cnt[0]));
               } else {
-                response.setStatusCode(500).setStatusMessage(reply.cause().getLocalizedMessage());
+                logger.error("Get users as a stream failed: " + reply.cause().getMessage(), reply.cause());
```
Comment (hjiebsco) [inline src/main/java/org/folio/rest/impl/UsersAPI.java]: RMB captured and logged it as ERROR.

Comment (julianladisch) [reply src/main/java/org/folio/rest/impl/UsersAPI.java]: Yes, you are right. I'm removing the duplicate logging.

File: src/main/java/org/folio/rest/impl/UsersAPI.java (hunk 1 above)
Comment (hjiebsco) [reply src/main/java/org/folio/rest/impl/UsersAPI.java]: Let's say the exception could happen (I do not know how), where are we going to handle it after the re-throw? Should there be another layer of try/catch to handle it and close the connection properly?

File: src/main/java/org/folio/rest/impl/UsersAPI.java [hunk 3]
```diff
@@ -183,13 +184,13 @@ public void getUsers(String query, String orderBy,
...
               } catch (JsonProcessingException e) {
-                logger.error("This error should never happen: " + e);
+                throw new UncheckedIOException(e);
               }
             }, reply -> {
               if (reply.succeeded()) {
                 response.write(String.format(JSON_USERS_FOOTER, cnt[0], cnt[0]));
               } else {
-                response.setStatusCode(500).setStatusMessage(reply.cause().getLocalizedMessage());
+                response.setStatusCode(500).setStatusMessage(reply.cause().getMessage());
```
Comment (hjiebsco) [inline src/main/java/org/folio/rest/impl/UsersAPI.java]: I noticed that in streaming mode, once we start to write response body, we can not change response status code anymore. HTTP/2 can do that, but we are not there yet. So the best we can do for now is to write the error message to the response body. So this line needs a bit tweak.

Comment (julianladisch) [reply src/main/java/org/folio/rest/impl/UsersAPI.java]: Yes. I've created https://issues.folio.org/browse/MODUSERS-169

**Pull request merged**